### DISCUSSION TITLE
Hill Shading

### DISCUSSION THREAD

**Discussion by baiti** [cgeo/cgeo#13671] [General]
The 11.13.2022 version of cgeo has hill shading. I have been using Hillshading with Locus Map (V3.x) before and it works like a charm. Once enabled, Locus asks for permission to download elevation data. Upon pressing "ok" it does that in the background, places whatever data is needed most likely into the private directories of the application. At least I cannot find any data via a file explorer.

I have downloaded all offline maps to `/Android/media/menion.android.locus.pro/maps/mapsVector` such that both, Locus and cgeo can use the same data for rendering maps, at least that is the idea and it is working for everything except hillshading.

Now hillshading in cgeo doesn't work even though it works in Locus. Most likely because Locus upon downloading the elevation data, does **not** place it into the diectory structure shown above. Ok so far so good. So I go into cgeo Settings -> Maps and click into the descriptive text at hillshading which leads me to the cgeo website which provides all kinds of explanation how hill shading is supposed to work. At some point I landed on `viewfinderpanoramas,org` and after some exhausting reading of a a lot of text I ultimately ended up downloading some zip file from a non secure website (viewfinderpamoramas.org). Then I moved and unpacked that .zip into the above path under a `_hgt` directory (that directory already existed, but was empty before). I restarted cgeo but still no hill shading.

The map is "Freizeitkarte", also the theme is freizeitkarte-v5. Locus shows shading, cgeo does not. Why is that so? 

And: Does it have to be so complicated? The process Locus has chosen to turn on hill shading is much much simpler. It is a single click on "OK" on a popup and that's it.

Update: A geocaching friend has managed to get it going. He says (a) doesn't work with Freizeitkarte and (b) with some other map, I assume osm, he says performance in cgeo's shading is  very weak and he has disabled the feature again.

I would assume performance in cgeo is bad in this scenario because the elevatio data is in the `/Android/media` path while Locus puts that into their private folders. In cgeo it is an all or nothng deal. Either everything is in the private folder or everything is under `Android/media/`

Last not least: Is it supposed to work all that way or is it just me not configuring it correctly?

**Comment by ztNFny** [comment]
> The map is "Freizeitkarte", also the theme is freizeitkarte-v5. Locus shows shading, cgeo does not. Why is that so?

https://github.com/cgeo/cgeo/issues/13538
tldr: Theme has to define the hillshading-layer which Freizeitkarte theme doesn't do.

It works in Locus because Locus is using their own fork of Mapsforge which handles hillshading completly different than Mapsforge.

> Does it have to be so complicated? The process Locus has chosen to turn on hill shading is much much simpler. It is a single click on "OK" on a popup and that's it.

auto-download is a Feature request already discussed. However that requires many things: Code to download, code to determine what needs to be downloaded, HGT file we have the permission to download from inside c:geo, a server from which we can download that much data, ...

> he says performance in cgeo's shading is very weak

what does "weak performance" mean? I see no difference in performance of map rendering with or without hillshading.

> Last not least: Is it supposed to work all that way or is it just me not configuring it correctly?

It's not just supposed to work, it works.

**Reply by baiti** [reply]
> > The map is "Freizeitkarte", also the theme is freizeitkarte-v5. Locus shows shading, cgeo does not. Why is that so?
> 
> #13538 tldr: Theme has to define the hillshading-layer which Freizeitkarte theme doesn't do.
> 
> It works in Locus because Locus is using their own fork of Mapsforge which handles hillshading completly different than Mapsforge.

Clearly and with no doubt superior

> 
> > Does it have to be so complicated? The process Locus has chosen to turn on hill shading is much much simpler. It is a single click on "OK" on a popup and that's it.
> 
> auto-download is a Feature request already discussed. However that requires many things: Code to download, code to determine what needs to be downloaded, HGT file we have the permission to download from inside c:geo, a server from which we can download that much data, ...

Well ... just SMOPs then.

> 
> > he says performance in cgeo's shading is very weak
> 
> what does "weak performance" mean? I see no difference in performance of map rendering with or without hillshading.
>

That I can't tell, would have to ask him for benchmark results.
 
> > Last not least: Is it supposed to work all that way or is it just me not configuring it correctly?
> 
> It's not just supposed to work, it works.

Of course it works. The question is different: "Is it supposed to work **that way**?" With "that way" meaning absolutely counter intuitive.

At this time I have decided to completely ignore this feature in cgeo and just switch over to Locus in those situations where I am hiking anywhere else than flat landscape.

**Reply by ztNFny** [reply]
> Clearly and with no doubt superior

I'm sure mapsforge is open for your commit of a better implementation :-P
